plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -545,7 +608,16 @@ public LocatedTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         boolean managed = table.get().managed();
 
         String tableLocation = table.get().location();
-        TableSnapshot tableSnapshot = getSnapshot(session, tableName, tableLocation, Optional.empty());
+        TableSnapshot tableSnapshot;
+        if (endVersion.isEmpty()) {
+            tableSnapshot = getCurrentSnapshot(session, tableName, tableLocation, Optional.empty());
+        }
+        else {
+            TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+            long version = getSnapshotIdFromVersion(fileSystem, tableLocation, endVersion.get());
+            tableSnapshot = getTimeTravelSnapshot(session, tableName, tableLocation, version);
```

Comment:
We should always use `getTimeTravelSnapshot`

Pls consider removing `getCurrentSnapshot`

Consider changing the record `QueriedTable` to the following form in order to accustom the use-case of calling the "current"/latest table snapshot.

```
QueriedTable(SchemaTableName schemaTableName, Optional<Long> version)
```